openc3-cmd-tlm-api/app/controllers/script_autocomplete_controller.rb

```diff
@@ -45,17 +54,38 @@ def get_ace_autocomplete_data
   end
 
   # private
+
+  def get_screen_keywords
+    OpenC3::MetaConfigParser.load(File.join(OpenC3::PATH, 'data', 'config', 'screen.yaml')).keys.sort
+  end
+
   def build_autocomplete_data(type, scope)
-    autocomplete_data = OpenC3::TargetModel.all(scope: scope).flat_map do |target_name, target_info|
-      OpenC3::TargetModel.packets(target_name, type: type.upcase.intern, scope: scope).flat_map do |packet|
-        packet_to_autocomplete_hashes(packet, target_info, type)
+    if type.upcase == 'SCREEN'
+      yaml = OpenC3::MetaConfigParser.load(File.join(OpenC3::PATH, 'data', 'config', 'screen.yaml'))
+      yaml.sort.map.each do |keyword, data|
+        params = []
+        if data['parameters']
+          params = data['parameters'].collect { |param| param['name'] }
+        end
+        # The snippet is what gets put in the file when you autocomplete
+        # Thus we put the keyword with all the parameters surround by <>
+        # e.g. SCREEN <Width> <Height> <Polling Period>
+        snippet = keyword.dup
+        snippet << " <#{params.join('> <')}>" unless params.empty?
+        {
+          :caption => keyword,
+          :snippet => snippet,
+          :meta => 'widget',
+        }
+      end
+    else
+      autocomplete_data = OpenC3::TargetModel.all(scope: scope).flat_map do |target_name, target_info|
+        OpenC3::TargetModel.packets(target_name, type: type.upcase.intern, scope: scope).flat_map do |packet|
+          packet_to_autocomplete_hashes(packet, target_info, type)
```

Comment:
This isn't going to scale well to lots of packets.

Reply:
Worked ok for 1000 cmd packets generated by loadsim